index.html

```diff
@@ -410,6 +463,94 @@ <h3 id="installation-sec">
       </h2>
       <section>
         <h3>
+          <code>BeforeInstallPromptEvent</code>
+        </h3>
+        <pre class="idl">
+enum AppBannerPromptOutcome {
+  "accepted",
+  "dismissed"
+};
+
+[Constructor(DOMString typeArg, optional BeforeInstallPromptEventInit eventInit)]
+interface BeforeInstallPromptEvent : Event {
+    readonly attribute Promise&lt;AppBannerPromptOutcome&gt; userChoice;
+    Promise&lt;void&gt; prompt();
+};
+
+dictionary BeforeInstallPromptEventInit : EventInit {
+  AppBannerPromptOutcome userChoice;
+};
+      </pre>
+        <p>
+          The <dfn><code>BeforeInstallPromptEvent</code></dfn> is dispatched
+          prior to activating an <a>automated install prompt</a>, allowing a
+          developer to prevent the default action for an install prompt.
+        </p>
+        <p>
+          Thus, the default action of the <a>BeforeInstallPromptEvent</a> is to
+          present an automated install prompt to the end user. Cancelling the
+          default action (via <code>.preventDefault()</code>) prevents the user
+          agent from presenting the automated install prompt.
+        </p>
+        <p>
+          To create an instance of the <code>BeforeInstallPromptEvent</code>
+          interface, run the <dfn>steps to construct a BeforeInstallPromptEvent
+          event</dfn>:
+        </p>
+        <ol>
+          <li>Let <var>userChoice</var> be null.
```

Comment:
These meta-variable names are confusing. *userChoicePromise* is `userChoice` (the `Promise<AppBannerPromptOutcome>`), and *userChoice* is the outcome that the promise is resolved with (the `AppBannerPromptOutcome`). That's messing with my head.

How about rename *userChoice* to *promptOutcome* (the outcome the promise is resolved with), and rename *userChoicePromise* to *userChoice* (the promise).